aviatrix/resource_aviatrix_arm_spoke_native_peering.go

```diff
@@ -0,0 +1,128 @@
+package aviatrix
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-aviatrix/goaviatrix"
+)
+
+func resourceAviatrixARMSpokeNativePeering() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAviatrixARMSpokeNativePeeringCreate,
+		Read:   resourceAviatrixARMSpokeNativePeeringRead,
+		Delete: resourceAviatrixARMSpokeNativePeeringDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"transit_gateway_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "",
+			},
+			"spoke_account_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "",
+			},
+			"spoke_region": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "",
+			},
+			"spoke_vpc_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "",
+			},
+		},
+	}
+}
+
+func resourceAviatrixARMSpokeNativePeeringCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*goaviatrix.Client)
+
+	armSpokeNativePeering := &goaviatrix.ArmSpokeNativePeering{
+		TransitGatewayName: d.Get("transit_gateway_name").(string),
+		SpokeAccountName:   d.Get("spoke_account_name").(string),
+		SpokeRegion:        d.Get("spoke_region").(string),
+		SpokeVpcID:         d.Get("spoke_vpc_id").(string),
+	}
+
+	log.Printf("[INFO] Creating Aviatrix arm spoke native peering: %#v", armSpokeNativePeering)
+
+	err := client.CreateArmSpokeNativePeering(armSpokeNativePeering)
+	if err != nil {
+		return fmt.Errorf("failed to create Aviatrix arm spoke native peering: %s", err)
+	}
+
+	d.SetId(armSpokeNativePeering.TransitGatewayName + "~" + armSpokeNativePeering.SpokeAccountName + "~" + armSpokeNativePeering.SpokeVpcID)
```

Comment:
Do we need account name ? I dont think we allow duplicated peering to same account name

Reply:
We do need account name. spoke_vpc_id is not unique. spoke_account_name puls spoke_vpc_id is unique.